### ISSUE TITLE
a bug leading sys_poll to be never blocked 

### ISSUE THREAD

**Issue by TheSayOL** [syswonder/ruxos#113]
`sys_poll()` is something like the codes below:

```rust
pub unsafe fn sys_poll(fds: *mut ctypes::pollfd, nfds: ctypes::nfds_t, timeout: c_int) -> c_int {
        loop {
            let fds_num = poll_all(fds)?;
            if fds_num > 0 {
                return Ok(fds_num as c_int);
            }
            crate::sys_sched_yield();
        }
}
```

that is, `sys_poll` won't block if the number of fd returned by `poll_all()` is bigger than 0. 

however, `poll_all()` is something like the codes below:

```rust
fn poll_all(fds: &mut [ctypes::pollfd]) -> LinuxResult<usize> {
    let mut events_num = 0;
    for pollfd_item in fds.iter_mut() {
        match get_file_like(intfd as c_int)?.poll() {
            Err(_) => { /* ... */ }
            Ok(state) => { /* do something with state ... */}
        }
        events_num += 1;
    }
    Ok(events_num)
}
```

that is, no matter what state the fd is, `events_num` always increases, which may cause the call of `sys_poll()` to return immediately but none of fds is ready.

a possible correction is, moving the `events_num+=1` to the expression of the `Ok` in the `match`, like the codes below:

```rust
fn poll_all(fds: &mut [ctypes::pollfd]) -> LinuxResult<usize> {
    let mut events_num = 0;

    for pollfd_item in fds.iter_mut() {
        let intfd = pollfd_item.fd;
        let events = pollfd_item.events;
        let revents = &mut pollfd_item.revents;
        match get_file_like(intfd as c_int)?.poll() {
            Err(_) => {
                if (events & ctypes::EPOLLERR as i16) != 0 {
                    *revents |= ctypes::EPOLLERR as i16;
                }
            }
            Ok(state) => {
                if state.readable && (events & ctypes::EPOLLIN as i16 != 0) {
                    *revents |= ctypes::EPOLLIN as i16;
                    events_num += 1;
                }

                if state.writable && (events & ctypes::EPOLLOUT as i16 != 0) {
                    *revents |= ctypes::EPOLLOUT as i16;
                    events_num += 1;
                }
            }
        }
    }
    Ok(events_num)
}
```

**Comment by coolyjg** [comment]
Correct this in your NEXT PR. Thanks!

**Comment by coolyjg** [comment]
Fixed in #122
